crates/e2e/src/backend_calls.rs

```diff
@@ -0,0 +1,229 @@
+use ink_env::Environment;
+use pallet_contracts_primitives::ContractInstantiateResult;
+use scale::{
+    Decode,
+    Encode,
+};
+
+use crate::{
+    builders::CreateBuilderPartial,
+    CallBuilderFinal,
+    CallDryRunResult,
+    CallResult,
+    ContractsBackend,
+    InstantiationResult,
+};
+
+use super::Keypair;
+
+pub struct CallBuilder<'a, E, Args, RetType, CB>
+where
+    E: Environment,
+    Args: Sync + Encode,
+    RetType: Send + Decode,
+
+    CB: ContractsBackend<E>,
+{
+    client: &'a mut CB,
+    caller: &'a Keypair,
+    message: &'a CallBuilderFinal<E, Args, RetType>,
+    value: E::Balance,
+    extra_gas_portion: Option<usize>,
+    storage_deposit_limit: Option<E::Balance>,
+}
+
+impl<'a, E, Args, RetType, CB> CallBuilder<'a, E, Args, RetType, CB>
+where
+    E: Environment,
+    Args: Sync + Encode,
+    RetType: Send + Decode,
+    E::Balance: Clone,
+
```

Comment:
nit: unnecessary empty line

Reply:
That was on purpose to logically separate generics